### PR TITLE
Add support for extra zimwriterfs arguments

### LINKED ISSUE #10: Extend zimscraperlib.zim to support more command arguments to zimwriterfs
Extend `ZimInfo` class to support more arguments as discussed here : https://github.com/openzim/sotoki/issues/107

### PR DESCRIPTION
**Pull request by satyamtg**
Fixes #10 

@rgaudin I have added support for extra zimwriterfs arguments in zimscraperlib.ziim and updated the tests accordingly. The code currently passes all tests. Have a look please.

### CODE REVIEW

**Comment by kelson42** [comment]
@satyamtg codefactor complains...

**Comment by satyamtg** [comment]
> @satyamtg codefactor complains...

Yup. Fixed that.

File: src/zimscraperlib/zim.py [hunk 1]
```diff
@@ -60,21 +73,38 @@ def to_zimwriterfs_args(self):
...
+
+        arg_list.extend(["--minChunkSize", str(minChunkSize)])
+
+        if self.redirects is not None:
+            arg_list.extend(["--redirects", self.redirects])
+        if self.source is not None:
+            arg_list.extend(["--source", self.source])
+        if self.flavour is not None:
+            arg_list.extend(["--flavour", self.flavour])
+        if self.tags != []:
```
Comment (rgaudin) [inline src/zimscraperlib/zim.py:95]: `if self.tags` should be enough

File: src/zimscraperlib/zim.py [hunk 2]
```diff
@@ -39,13 +42,23 @@ def __init__(
...
 
-    def to_zimwriterfs_args(self):
-        return [
+    def to_zimwriterfs_args(
+        self,
+        verbose=False,
+        inflateHtml=False,
+        uniqueNamespace=False,
+        withoutFTIndex=False,
+        minChunkSize=2048,
```
Comment (rgaudin) [inline src/zimscraperlib/zim.py:59]: we don't want to set that. If we don't supply a value, we should default to not supply it to zimwriterfs and leave it up to the tool to set it. So use None.

File: src/zimscraperlib/zim.py [hunk 3]
```diff
@@ -39,13 +42,23 @@ def __init__(
         self.name = name
         self.tags = tags
         self.scraper = scraper
+        self.source = source
+        self.flavour = flavour
+        self.redirects = redirects
```
Comment (rgaudin) [inline src/zimscraperlib/zim.py:47]: Redirects is a zimwriterfs feature (a path to a file). If should not be stored here and passed as arg to the other function.

**Comment by rgaudin** [comment]
Thank you, I've made a few minor inline comments. I think the CI is misconfigured for external PRs… I'll try to fix that. Make sure coverage stays at 100% (haven't tested yet).

File: src/zimscraperlib/zim.py (hunk 3 above)
Comment (satyamtg) [reply src/zimscraperlib/zim.py:47]: Okay. Will change that.

File: src/zimscraperlib/zim.py (hunk 2 above)
Comment (satyamtg) [reply src/zimscraperlib/zim.py:59]: I just put 2048 as it is the default value that zimwriterfs uses. Although it shouldn't be a problem but I agree that we should not add this to the argument list unless a value is provided. Will set it to None and add it to arguments only if a value is provided.

File: src/zimscraperlib/zim.py (hunk 1 above)
Comment (satyamtg) [reply src/zimscraperlib/zim.py:95]: Yup. Actually, it would be faster too.

**Comment by satyamtg** [comment]
@rgaudin did the changes and updated the test to be more robust. It should have 100% code coverage.

**Comment by rgaudin** [comment]
@satyamtg tank you ; can you merge the master branch in and push so I can merge this? As the tests are misconfigured, github can't do that automatically.

**Comment by satyamtg** [comment]
@rgaudin, okay. Doing that.

**Pull request merged**